dart-impl/base/src/logging.c

```diff
@@ -22,3 +22,29 @@ const int dart__base__unit_term_colors[DART_LOG_TCOL_NUM_CODES-1] = {
   DART_LOG_TCOL_RED,
   DART_LOG_TCOL_BLUE
 };
+
+
+enum dart__base__logging_loglevel
+dart__base__logging_env_loglevel()
+{
+  static enum dart__base__logging_loglevel level = DART_LOGLEVEL_TRACE;
+  static int log_level_parsed = 0;
+  if (!log_level_parsed) {
+    const char *envstr = getenv(DART_LOGLEVEL_ENVSTR);
+    if (envstr) {
```

Comment:
I've seen the `dart_config` 'interface', which is horrible as it exposes an internal data structure to the world. This might be OK for setting a single value but it breaks pretty much every interface design rule. 

Moreover, considering that DART is meant to be used as a stand-alone runtime system, there should be means of controlling the log level without changing client code, i.e., without having to call DART functions at runtime. I do not see what is wrong with calling `getenv` in DART.  We can implement a way to override the environment setting (if any) through a re-designed `dart_config` interface, though.